pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -1848,25 +1928,49 @@ impl Timeline {
             lowest_retain_lsn
         );
         // Step 1: (In the future) construct a k-merge iterator over all layers. For now, simply collect all keys + LSNs.
-        // Also, collect the layer information to decide when to split the new delta layers.
+        // Also, verify if the layer map can be split by drawing a horizontal line at every LSN start/end split point.
+        let mut lsn_split_point = BTreeSet::new();
         let mut downloaded_layers = Vec::new();
-        let mut delta_split_points = BTreeSet::new();
         for layer in &layer_selection {
             let resident_layer = layer.download_and_keep_resident().await?;
             downloaded_layers.push(resident_layer);
```

Comment:
Youy can push down the downloaded_layers business until after this `for` loop. 

This has the advantage of us not downloading a lot of layers, only to discover later that we can't run the bottom-most compaction because the layer map doesn't have the LSN split property.